libopae/plugins/xfpga/sysobject.c

```diff
@@ -119,6 +120,23 @@ fpga_result __FPGA_API__ xfpga_fpgaObjectGetObject(fpga_object parent, const cha
 	return make_sysfs_object(objpath, name, object, flags, _obj->handle);
 }
 
+fpga_result __FPGA_API__ xfpga_fpgaObjectGetObjectAt(fpga_object parent,
+						     size_t idx,
+						     fpga_object *object)
+{
+	ASSERT_NOT_NULL(parent);
+	ASSERT_NOT_NULL(object);
+	struct _fpga_object *_obj = (struct _fpga_object *)parent;
+	if (_obj->type == FPGA_SYSFS_FILE) {
```

Comment:
I think we should grab the lock inside the fpga_object when we are introspecting or modifying the object.  This would apply to each of the xfpga_ api's.

Reply:
I'm not sure about getting child elements from a container type object since the number of items in the container won't change after the container object has been created. I agree with reading/writing from the driver though.

Reply:
Don't we need to prevent the object from being destroyed while we introspect..  I guess in that scenario, we would be accessing invalid pointers after the destruction occurred.  I think this is what the handles, etc, do though.  I'd feel safer about it if we followed that precedent throughout.